web_src/js/utils.ts

```diff
@@ -32,16 +32,18 @@ export function stripTags(text: string): string {
 }
 
 export function parseIssueHref(href: string): IssuePathInfo {
+  // FIXME: it should use pathname and trim the appSubUrl ahead
   const path = (href || '').replace(/[#?].*$/, '');
   const [_, ownerName, repoName, pathType, indexString] = /([^/]+)\/([^/]+)\/(issues|pulls)\/([0-9]+)/.exec(path) || [];
   return {ownerName, repoName, pathType, indexString};
 }
 
-export function parseIssueNewHref(href: string): IssuePathInfo {
-  const path = (href || '').replace(/[#?].*$/, '');
-  const [_, ownerName, repoName, pathTypeField] = /([^/]+)\/([^/]+)\/(issues\/new|compare\/.+\.\.\.)/.exec(path) || [];
-  const pathType = pathTypeField ? (pathTypeField.startsWith('issues/new') ? 'issues' : 'pulls') : undefined;
-  return {ownerName, repoName, pathType};
+export function parseRepoOwnerPathInfo(pathname: string): RepoOwnerPathInfo {
+  const appSubUrl = window.config.appSubUrl;
+  if (appSubUrl && pathname.startsWith(appSubUrl)) pathname = pathname.substring(appSubUrl.length);
+  const path = (pathname || '').replace(/[#?].*$/, '');
```

Comment:
two small problem to point out

- may we should trust types and not null check pathname. to use empty string can hide callers issue
- when it is not a repo path, we returns empty object and check if ownerName is exists even though it's type does not say it can be nullable. may we can return null instead null ownerName?

Reply:
I didn't do that for 2 reasons:

1. There was a long argument about returning null or not: https://github.com/go-gitea/gitea/pull/29123#pullrequestreview-1881660139
2. The `parseIssueHref` was already written that way: https://github.com/go-gitea/gitea/pull/17398#issuecomment-949559015

To avoid unnecessary arguments and changes, I didn't use `null` here.